File: LaTeXDraw/src/net/sf/latexdraw/actions/DrawingAction.java
package net.sf.latexdraw.actions;

import org.malai.action.Action;

import net.sf.latexdraw.glib.models.interfaces.IDrawing;

/**
 * This abstract action uses a drawing.<br>
 * <br>
 * This file is part of LaTeXDraw.<br>
 * Copyright (c) 2005-2012 Arnaud BLOUIN<br>
 * <br>
 * LaTeXDraw is free software; you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation; either version 2 of the License, or (at your option) any later version.
 * <br>
 * LaTeXDraw is distributed without any warranty; without even the implied
 * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * General Public License for more details.<br>
 * <br>
 * 01/07/2010<br>
 * @author Arnaud BLOUIN
 * @since 3.0
 */
public abstract class DrawingAction extends Action {
	/** The drawing that will be handled by the action. */
	protected IDrawing drawing;
	
	
	/**
	 * Creates the action.
	 * @since 3.0
	 */
	public DrawingAction() {
		super();
	}


	@Override
	public boolean canDo() {
		return drawing!=null;
	}
	
	
	@Override
	public void flush() {
		super.flush();
		drawing = null;
	}
	
	
	/**
	 * @param drawing The drawing that will be handled by the action
	 * @since 3.0
	 */
	public void setDrawing(final IDrawing drawing) {
		this.drawing = drawing;
	}
	
	
	/**
	 * @return The drawing that will be handled by the action
	 * @since 3.0
	 */
	public IDrawing getDrawing() {
		return drawing;
	}
}

File: latexdraw-core/net.sf.latexdraw/src/main/net/sf/latexdraw/actions/DrawingAction.scala
/*
  * This file is part of LaTeXDraw.
  * Copyright (c) 2005-2014 Arnaud BLOUIN
  * LaTeXDraw is free software; you can redistribute it and/or modify it under
  * the terms of the GNU General Public License as published by the Free Software
  * Foundation; either version 2 of the License, or (at your option) any later version.
  * LaTeXDraw is distributed without any warranty; without even the implied
  * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
  * General Public License for more details.
 */
package net.sf.latexdraw.actions

import java.util.Optional

import net.sf.latexdraw.models.interfaces.shape.IDrawing

/**
  * This trait encapsulates a drawing attribute.
  */
trait DrawingAction {
	/** The drawing that will be handled by the action. */
	var _drawing: Optional[IDrawing] = Optional.empty()


	/**
	  * @param drawing The drawing that will be handled by the action
	  * @since 3.0
	  */
	def setDrawing(drawing: IDrawing) {
		_drawing = Optional.ofNullable(drawing)
	}


	/**
	  * @return The drawing that will be handled by the action
	  * @since 3.0
	  */
	def drawing = _drawing
}
